src/pal/src/init/pal.cpp

```diff
@@ -658,12 +658,14 @@ PAL_InitializeCoreCLR(
     int result = PAL_Initialize(1, &szExePath);
 #endif // __APPLE__
     if (result != 0)
+    {
         return GetLastError();
+    }
 
     // Now that the PAL is initialized it's safe to call the initialization methods for the code that used to
     // be dynamically loaded libraries but is now statically linked into CoreCLR just like the PAL, i.e. the
     // PAL RT and mscorwks.
-    if (!LOADInitCoreCLRModules())
+    if (!LOADInitCoreCLRModules(g_szCoreCLRPath))
```

Comment:
I would like to get rid of the global.  Using globals to pass parameters is horrible and not very clean.

From: Jan Vorlicek [mailto:notifications@github.com]
Sent: Wednesday, April 22, 2015 1:37 PM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Share hardware exception between coreclr and DAC (#771)


In src/pal/src/init/pal.cpp<https://github.com/dotnet/coreclr/pull/771#discussion_r28912154>:

>

>      // Now that the PAL is initialized it's safe to call the initialization methods for the code that used to

>      // be dynamically loaded libraries but is now statically linked into CoreCLR just like the PAL, i.e. the

>      // PAL RT and mscorwks.

> -    if (!LOADInitCoreCLRModules())

> +    if (!LOADInitCoreCLRModules(g_szCoreCLRPath))

I wonder in the change that added the parameter to this function is necessary. We cannot get rid of the g_szCoreCLRPath anyways, since it is being used at another place to bind string resource domain to the path. So it seems we could use this global inside of the LOADInitCoreCLRModules.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/771/files#r28912154>.